client/jetpack-connect/controller.js

```diff
@@ -359,3 +359,25 @@ export function sso( context, next ) {
 	);
 	next();
 }
+
+export function authorizeOrSignup( context, next ) {
+	const isLoggedIn = isUserLoggedIn( context.store.getState() );
+
+	if ( isLoggedIn ) {
+		authorizeForm( context, next );
+		return;
+	}
+
+	signupForm( context, next );
+}
+
+export function redirectToLoginIfLoggedOut( context, next ) {
+	const isLoggedIn = isUserLoggedIn( context.store.getState() );
+
+	if ( ! isLoggedIn ) {
+		page( login( { isNative: true, isJetpack: true, redirectTo: context.path } ) );
+		return;
+	}
+
+	next();
+}
```

Comment:
This very universal helper can be already imported from `import { redirectLoggedOut } from 'calypso/controller'`.

Could we somehow reuse it, after figuring out how to pass the `isJetpack` option?

Reply:
`redirectLoggedOut` seems to be doing a bit more than this one is doing. I'd prefer to keep them separate, especially considering the purpose of this PR.